src/core/uint.js

```diff
@@ -108,16 +110,32 @@ UInt.prototype.copyTo = function(d) {
   return d;
 };
 
-UInt.prototype.equals = function(d) {
-  return this.is_valid() && d.is_valid() && this._value.equals(d._value);
+UInt.prototype.equals = function(o) {
+  return this.is_valid() &&
+         o.is_valid() &&
+         // This throws but the expression will short circuit
+         this.cmp(o) === 0;
+};
+
+UInt.prototype.cmp = function(o) {
+  assert(this.is_valid() && o.is_valid());
+  return this._value.ucmp(o._value);
```

Comment:
They are UInt, so it's fine. I think I maybe have been confused that it was required when I used it. `u` means u)nsigned comparison. cmp actually calls it, after checking `sign`